CHANGELOG.md

```diff
@@ -0,0 +1,178 @@
+      - `ForceCurrent(PinSiteData<DCPowerSourceSettings> settings, bool waitForSourceCompletion = false)`
+    - Overloads added for the `Digital.PPMU.ForceVoltage` Extension Method
+      - `void ForceVoltage(IDictionary<string, double> voltageLevels, double? currentLimitRange = null, double? apertureTime = null, double? settlingTime = null)`
+      - `ForceVoltage(PPMUSettings settings)`
+      - `ForceVoltage(SiteData<PPMUSettings> settings`
+      - `ForceVoltage(PinSiteData<PPMUSettings> settings)`
+    - Overloads added for the `Digital.PPMU.ForceCurrent` Extension Method
+      - `ForceCurrent(PPMUSettings settings)`
+      - `ForceCurrent(SiteData<PPMUSettings> settings)`
+      - `ForceCurrent(PinSiteData<PPMUSettings> settings)`
+    - Overloads added for querying Session Bundle (Digital, DCPower, DAQmx, DMM, Scope, Fgen, Sync) with boolean parameter to enable whether to automatically filter out non-instrument-related pins. Note, these new methods will return null if all pins are non-instrument-related rather than throwing an exception.
+      - `TSMSessionManager.Digital(string pin, bool filterPins)`
+      - `TSMSessionManager.Dmm(string pin, bool filterPins)`
+      - `TSMSessionManager.DAQmx(string pin, bool filterPins)`
+      - `TSMSessionManager.Scope(string pin, bool filterPins)`
+      - `TSMSessionManager.Fgen(string pin, bool filterPins)`
+      - `TSMSessionManager.Sync(string pin, bool filterPins)`
+  - **Utilities**
+    - New `FilterPinsOrPinGroups` extension method added that extends `ISemiconductorModuleContext` to filter pins or pin groups by instrument type, and provide a List of the filtered pins (pin groups intact) and the index of each of the filtered pins in the input pins or pin groups array, as well as a flatted list of pins (pin groups expanded).
+      - `FilterPinsOrPinGroups(string[] pinsOrPinGroups, string instrumentTypeId, out IList<string[]> pins, out IList<int> pinIndexes, out string[] pinsFlattened)`
+        - Return Type: `void`
+
+- ### Changed
+  
+  - Math Operations methods now support most common numeric data types with improved exceptions for catching unsupported types and input array size mismatches.
+  - Math Operations methods now support scalar input values when the underlying type, `T`, of the `SiteData<T>` or `PinSiteData<T>` object is an array type.
+    - For `PinSiteData<T>` objects, the scalar value will be applied to all elements in the array, across each pin and site.
+    - For `SiteData<T>` objects, the scalar value will be applied to all elements in the array, across each site.
+  - `Utilities.TryDeterminePowerLineFrequency` updated to now support OfflineMode.
+  - `TestStandSteps.ContinuityTest` modified to correctly accept negative current level values.
+  - CSProject files for TestStandSteps and Extensions now exclude net48 path from being included as a folder within the project, which was could cause build issues for contributors in certain situations.
```

Comment:
"..., which was causing..." or "..., which could cause..."?

Reply:
@lornazh, thanks for catching this! It should just be "which could cause...". I will make this change now.